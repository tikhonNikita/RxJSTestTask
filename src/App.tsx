import React, { useState, useEffect } from 'react';
import { Observable } from 'rxjs';

import IData from './types/IData';
import configurableSubscriber from './observable/observable';
import { Dashboard } from './components/Dashboard/Dashboard';

import './App.css';

const initialDisplayEmitting = (data: IData) => {
  return (
    data.temperature !== '' && data.pressure !== '' && data.humidity !== ''
  );
};

const observable = new Observable(configurableSubscriber({}));

function App() {
  const [dimensions, setDimensions] = useState({
    temperature: '',
    pressure: '',
    humidity: ''
  });

  useEffect(() => {
    observable.subscribe(dimensionsData => {
      setDimensions(dimensionsData as IData);
    });
  }, []);

  return (
    <div className="App">
      {initialDisplayEmitting(dimensions) && (
        <Dashboard {...dimensions}></Dashboard>
      )}
    </div>
  );
}

export default App;
