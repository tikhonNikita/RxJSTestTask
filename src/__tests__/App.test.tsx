import React from 'react';
import { create } from 'react-test-renderer';

import App from '../App';

test('renders app component', () => {
  const app = create(<App />);

  expect(app.toJSON()).toMatchSnapshot();
});
