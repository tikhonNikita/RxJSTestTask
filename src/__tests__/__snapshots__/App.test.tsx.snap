// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component 1`] = `
<div
  className="App"
/>
`;
